### DISCUSSION TITLE
Custom validators not works?

### DISCUSSION THREAD

**Discussion by qarmin** [rwf2/Rocket#2282] [Questions]
I have code rest endpoint which takes JSON<NewPatientInfo> as argument
```
#[post("/patients", data = "<new_patient>")]
pub async fn create_patient(
    db: &State<DatabaseConnection>,
    new_patient: Json<NewPatientInfo>,
) -> Result<Json<PatientInfo>, ErrorResponder> 
```
with field birth date and custom validator
```
#[derive(Serialize, Deserialize, JsonSchema, FromForm)]
pub struct NewPatientInfo {
    #[field(validate = crate::commons::rocket_validators::date_validator())]
    pub(crate) birth_date: String,
```
Validator is simple(and useless), but not works - there is no SSSSS in output
```
pub fn date_validator<'v>(str_date: &str) -> form::Result<'v, ()> {
    println!("SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS");
    Ok(())
}
```
When I change ` crate::commons::rocket_validators::date_validator` to invalid path, I have compilation error, so I'm sure that this points to correct function, but still I don't know why this not works

**Comment by SergioBenitez** [comment]
The `field` struct field attribute is used by the `FromForm` derive. If you were to accept a `Form<NewPatientInfo>`, the validation routine would run. Instead you're accepting a `Json`, which uses `Deserialize`. If you need extra validation for fields, you'll need to go through serde.
